File: src/storage_security/python/secure_data_storage.py
from cryptography.hazmat.primitives.ciphers import Cipher, algorithms, modes
from cryptography.hazmat.primitives import padding
from cryptography.hazmat.backends import default_backend
import os

def encrypt(data, key):
    cipher = Cipher(algorithms.AES(key), modes.CBC(os.urandom(16)), backend=default_backend())
    encryptor = cipher.encryptor()
    padder = padding.PKCS7(algorithms.AES.block_size).padder()
    padded_data = padder.update(data.encode()) + padder.finalize()
    return cipher.encryptor().update(padded_data) + cipher.encryptor().finalize()

def decrypt(encrypted_data, key):
    cipher = Cipher(algorithms.AES(key), modes.CBC(encrypted_data[:16]), backend=default_backend())
    decryptor = cipher.decryptor()
    unpadder = padding.PKCS7(algorithms.AES.block_size).unpadder()
    decrypted_data = unpadder.update(decryptor.update(encrypted_data[16:]) + decryptor.finalize())
    return decrypted_data.decode()

if __name__ == "__main__":
    key = os.urandom(16)
    sensitive_data = "SecurePassword123"

    encrypted_data = encrypt(sensitive_data, key)
    print(f"Encrypted Data: {encrypted_data}")

    decrypted_data = decrypt(encrypted_data, key)
    print(f"Decrypted Data: {decrypted_data}")

File: src/storage_security/csharp/SecureDataStorage.cs
using System;
using System.IO;
using System.Security.Cryptography;
using System.Text;

class SecureDataStorage {
    public static string Encrypt(string plainText, byte[] key, byte[] iv) {
        using (Aes aes = Aes.Create()) {
            aes.Key = key;
            aes.IV = iv;
            using (var encryptor = aes.CreateEncryptor(aes.Key, aes.IV)) {
                using (var ms = new MemoryStream()) {
                    using (var cs = new CryptoStream(ms, encryptor, CryptoStreamMode.Write)) {
                        using (var writer = new StreamWriter(cs)) {
                            writer.Write(plainText);
                        }
                    }
                    return Convert.ToBase64String(ms.ToArray());
                }
            }
        }
    }

    public static string Decrypt(string cipherText, byte[] key, byte[] iv) {
        using (Aes aes = Aes.Create()) {
            aes.Key = key;
            aes.IV = iv;
            using (var decryptor = aes.CreateDecryptor(aes.Key, aes.IV)) {
                using (var ms = new MemoryStream(Convert.FromBase64String(cipherText))) {
                    using (var cs = new CryptoStream(ms, decryptor, CryptoStreamMode.Read)) {
                        using (var reader = new StreamReader(cs)) {
                            return reader.ReadToEnd();
                        }
                    }
                }
            }
        }
    }

    public static void Main() {
        string sensitiveData = "SensitiveData123";
        byte[] key = Aes.Create().Key;
        byte[] iv = Aes.Create().IV;

        string encryptedData = Encrypt(sensitiveData, key, iv);
        Console.WriteLine("Encrypted Data: " + encryptedData);

        string decryptedData = Decrypt(encryptedData, key, iv);
        Console.WriteLine("Decrypted Data: " + decryptedData);
    }
}